src/components/SponsorPage/SponsorPage.jsx

```diff
@@ -77,32 +94,113 @@ export function SponsorPage() {
         );
     }
 
-    // const propsTalentData = {
-    //     profile,
-    //     editting,
-    //     firstName,
-    //     setFirstName,
-    //     lastName,
-    //     setLastName,
-    // };
+    function save() {
+        talentDataRef.current?.validate();
+        let isValid = talentDataRef.current?.validate();
+        if (isValid) {
+            const obj = {
+                first_name: firstName.name,
+                last_name: lastName.name,
+                count_of_kudos: numberKudos.kudos,
+            };
 
-    // const propsAbout = {
-    //     profile,
-    //     editting,
-    //     setEditting,
-    //     setModalIsOpen,
-    //     setCancelModalIsOpen,
-    //     saveError,
-    // };
+            try {
+                TalentsService.editSponsor(user.id, obj, token)
+                    .then(() => {
+                        setSaveError("");
+                        setProfile((prev) => ({
+                            ...prev,
+                            first_name: firstName.name,
+                            last_name: lastName.name,
+                            amount: numberKudos.kudos,
+                        }));
+                        setKudos(kudos + parseInt(numberKudos.kudos));
```

Comment:
use previous state to change this constant
```suggestion
                        setKudos((prev)=> prev + parseInt(numberKudos.kudos));
```